tool/tsh/common/tsh.go

```diff
@@ -1817,8 +1817,23 @@ func onLogin(cf *CLIConf) error {
 
 	// If the cluster is using single-sign on, providing the user name
 	// with --user is likely a mistake, so display a warning.
-	if cf.AuthConnector != "" && cf.AuthConnector != constants.LocalConnector && cf.Username != "" {
-		fmt.Fprintf(os.Stderr, "WARNING: Ignoring Teleport user (%v) for Single Sign-On (SSO) login.\nProvide the user name during the SSO flow instead. Use --auth=local if you did not intend to login with SSO.\n", cf.Username)
+	if cf.Username != "" {
+		var displayIgnoreUserWarning = false
+		if cf.AuthConnector != "" && cf.AuthConnector != constants.LocalConnector {
+			displayIgnoreUserWarning = true
+		} else if cf.AuthConnector == "" {
+			// Get the Ping so we check if the default Auth type is SSO
+			pr, err := tc.Ping(cf.Context)
```

Comment:
This introduces an extra ping (and thus another proxy round trip) in the login process.

Can we avoid this? We've done some work to improve `tsh login` latency and adding another ping will erase some of he progress we've made.

Reply:
@zmb3 I did review this initially as a concern myself and should have included that in the PR comments. The Ping call is cached so it would not add network activity.  See https://github.com/gravitational/teleport/blob/d9ea2061d79b6ceb9484f423c4922836130db9c0/lib/client/api.go#L3371

Reply:
See https://github.com/gravitational/teleport/blob/d9ea2061d79b6ceb9484f423c4922836130db9c0/lib/client/api.go#L4154-L4156

Reply:
TIL

Reply:
ok to approve @zmb3 ?  I set it to attempt to do as little as possible in compute steps.